lib/motion/command/android_setup.rb

```diff
@@ -0,0 +1,167 @@
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+module Motion; class Command
+  class AndroidSetup < Command
+    DefaultDirectory = File.join(Dir.home, 'rubymotion-android-test')
+    DefaultSDKVersion = '24.1.2'
+    DefaultNDKVersion = 'r10d'
+
+    DL_GOOGLE = "https://dl.google.com/android"
+
+    # use `tools/android list sdk --extended --all`
+    # to get a list of available packages
+    PACKAGES_LIST = %w{
+      platform-tools
+      build-tools-22.0.0
+      android-22
+      system-image
+      extra-android-support
+    }
+
+    self.summary = 'Setup android.'
+    self.description = 'Setup the android environnement : sdk/ndk.'
+
+    def initialize(argv)
+      @directory = argv.option('directory') || DefaultDirectory
+      @sdk_version = argv.option('sdk_version') || DefaultSDKVersion
+      @ndk_version = argv.option('ndk_version') || DefaultNDKVersion
+      @ndk_directory = File.join(@directory, 'ndk')
+      @sdk_directory = File.join(@directory, 'sdk')
+      @tmp_directory = File.join(@directory, 'tmp')
+      super
+    end
+
+    def self.options
+      [
+        ['--directory=[PATH]', "The android install directory."],
+        ['--sdk_version=[VERSION]', "The android SDK version."],
+        ['--ndk_version=[VERSION]', "The android NDK version."]
+      ].concat(super)
+    end
+
+    def run
+      check_java
+      create_directory_structure
+      setup_sdk
+      setup_ndk
+      check_env_variables
+      open_gui
+    end
+
+    protected
+
+    def open_gui
+      system(android_executable)
+    end
+
+    def android_executable
+      File.join(@sdk_directory, 'tools', 'android')
+    end
+
+    def setup_ndk
+      $stderr.puts("-- Setup NDK...")
+      if !File.exist?(@ndk_directory) || @ndk_version != current_ndk_version
+        $stderr.puts("Installing NDK : version #{@ndk_version}")
+        ndk_url = File.join(DL_GOOGLE, 'ndk', "android-ndk-#{@ndk_version}-darwin-x86_64.bin")
+        ndk_bin = File.join(@tmp_directory, 'ndk.bin')
+        download(ndk_url, ndk_bin)
+        system("chmod +x #{ndk_bin}")
+        system("cd #{@tmp_directory} && #{ndk_bin}")
+        FileUtils.mv(File.join(@tmp_directory, "android-ndk-#{@ndk_version}"), @ndk_directory)
+      else
+        $stderr.puts("Installed NDK is up to date.")
+      end
+    end
+
+    def current_ndk_version
+      file = File.open(File.join(@ndk_directory, 'RELEASE.TXT'), "rb")
+      version = file.read.strip.split(' ')[0]
+      file.close
```

Comment:
Could be replaced with:

`version = File.read(File.join(@ndk_directory, 'RELEASE.TXT'). strip.split(' ')[0]`